packages/@aws-cdk/cdk/lib/token.ts

```diff
@@ -19,39 +46,58 @@ export const RESOLVE_METHOD = 'resolve';
  * Tokens can be embedded into strings while retaining their original
  * semantics.
  */
-export class Token {
+export abstract class Token implements IToken {
   /**
-   * @deprecated use `Token.isToken`
+   * Return true if input value is a Token or encoded Token
    */
   public static unresolved(obj: any): boolean {
     return unresolved(obj);
   }
 
   /**
-   * Returns true if obj is a token (i.e. has the resolve() method or is a
-   * string or array which includes token markers).
+   * Returns true if obj is an IToken object.
    *
    * @param obj The object to test.
    */
-  public static isToken(obj: any): boolean {
-    return unresolved(obj);
+  public static isToken(obj: any): obj is IToken {
```

Comment:
I feel `Token.isToken(x)` should return true if `x` is a token or a token encoded as a primitive (list/string/number). 
The test of whether something implements `IToken` should probably not even be a public API.